src/plugins/func/funcUtils.ts

```diff
@@ -0,0 +1,99 @@
+import yaml from "js-yaml";
+import fs from "fs";
+
+const functionsRegex = /functions:([\s\S]*?)\n\n/g
+
+export class FuncPluginUtils {
+
+  public static getServerlessYml() {
+    return fs.readFileSync("serverless.yml", "utf-8");
+  }
+
+  public static getFunctionsYml(serverlessYml?: string) {
+    serverlessYml = serverlessYml || FuncPluginUtils.getServerlessYml();
+    const functionsSection = serverlessYml.match(functionsRegex)[0];
+    return yaml.safeLoad(functionsSection);
+  }
+
+  public static updateFunctionsYml(functionYml: any, serverlessYml?: string) {
+    serverlessYml = serverlessYml || FuncPluginUtils.getServerlessYml();
+    const newFunctionsYaml = yaml.dump(functionYml);
+    const newServerlessYaml = serverlessYml.replace(functionsRegex, `${newFunctionsYaml}\n`);
+    fs.writeFileSync("serverless.yml", newServerlessYaml);
+  }
+
+  public static getFunctionHandler(name: string) {
+    return `"use strict";
```

Comment:
So I've spent some time trying to make this work. My problem has been what kind of file to put the template in. Here are the options I've tried:

Template file options:
1. TypeScript file
   - Gets transpiled to JavaScript in build - doesn't keep template structure
2. JavaScript file
   - Can't be included in build
   - Tried setting 'allowJs' to be true, but that conflicts with 'declarations' being true
3. Text file
   - Could make it work, but it would mean an extra step post-build of copying the text file over. Not sure that's warranted just to avoid the inline script

Other options?

Reply:
Yah, the ideal solution would be to be able to include a js file of the handler since that's technically what we're packaging, but if that isn't really an option, then inline seems reasonable.

Reply:
Let's add a work item to re-review this later.

Reply:
[AB#18775]